grails-app/services/org/broadinstitute/orsp/IssueService.groovy

```diff
@@ -243,6 +281,89 @@ class IssueService implements UserInfo {
         issue
     }
 
+    /**
+     * Handles the update of project key to an existing issue.
+     *
+     * @param issue The existing issue to update.
+     * @param input Map of form arguments that come from any of the various update forms
+     * @return Persisted issue
+     */
+    @Transactional
+    Issue updateProjectkey(Issue issue, Map<String, Object> input) throws DomainException {
+        // update projectKey and type if response to determination questions changed
+            def issueType = IssueType.valueOfPrefix(input.get("type"))
```

Comment:
Prefer fully typed objects over anonymous `def`s. Similar feedback throughout this method.